involute_gears.scad

```diff
@@ -42,488 +42,538 @@ pi=3.1415926535897932384626433832795;
+            
+            //add in spokes
+            if (spokes>0) 
+                for(i=[0:spokes-1])
+                    translate([0,0,rim_thickness/2])
+                    rotate([90,0,i*360/spokes])
+                    translate([0,0,spoke_radius])
+                    {
+                        if (spoke_square==true){
+                             resize([spoke_width,spoke_thickness,spoke_length])
+                            translate([0,0,.5])
+                            cube(1,center=true);
+                        }
+                        if (spoke_square==false){
+                            resize([spoke_width,spoke_thickness,spoke_length])
+                            cylinder(h=1,d=1);
+        }
+                    }
+        }
+        
+        //remove the center bore
+        translate ([0,0,-1])
+        linear_extrude_flat_option(flat =flat, height=2+max(rim_thickness,hub_thickness,gear_thickness))
+        circle (r=bore_diameter/2);
+        
+        //remove circles from gear body
+        if (circles>0)
+        {
+            for(i=[0:circles-1])
+                rotate([0,0,i*360/circles])
+                translate([circle_orbit_diameter/2,0,-1])
+                linear_extrude_flat_option(flat =flat, height=max(gear_thickness,rim_thickness)+3)
+                circle(r=circle_diameter/2);
+        }
+    }
 }
 
 module rack(
-		number_of_teeth=15,
-		circular_pitch=false, diametral_pitch=false,
-		pressure_angle=28,
-		clearance=0.2,
-		rim_thickness=8,
-		rim_width=5,
-		flat=false)
+        number_of_teeth=15,
+        circular_pitch=false, diametral_pitch=false,
+        pressure_angle=28,
+        clearance=0.2,
+        rim_thickness=8,
+        rim_width=5,
+        flat=false)
 {
 
-	if (circular_pitch==false && diametral_pitch==false)
-		echo("MCAD ERROR: gear module needs either a diametral_pitch or circular_pitch");
-
-	//Convert diametrial pitch to our native circular pitch
-	circular_pitch = (circular_pitch!=false?circular_pitch:180/diametral_pitch);
-	pitch = circular_pitch / 180 * PI;
-
-	addendum = circular_pitch / 180;
-	dedendum = addendum + clearance;
-	pitch_slope = tan(pressure_angle);
-
-	linear_extrude_flat_option(flat=flat, height=rim_thickness)
-		union()
-		{
-			translate([0,-dedendum-rim_width/2])
-				square([number_of_teeth*pitch, rim_width],center=true);
-
-			p1 = pitch / 4 + pitch_slope * dedendum;
-			p2 = pitch / 4 - pitch_slope * addendum;
-			for(i=[1:number_of_teeth])
-				translate([pitch*(i-number_of_teeth/2-0.5),0])
-					polygon(points=[
-							[-p1,-dedendum],
-							[p1,-dedendum],
-							[p2,addendum],
-							[-p2,addendum]
-					]);
-		}
+    if (circular_pitch==false && diametral_pitch==false)
+        echo("MCAD ERROR: gear module needs either a diametral_pitch or circular_pitch");
+
+    //Convert diametrial pitch to our native circular pitch
+    circular_pitch = (circular_pitch!=false?circular_pitch:pi/diametral_pitch);
```

Comment:
Can't accept this change. I know the current version is broken in the handling of circular pitch, but changing it will break all the designs using this module right now.

Reply:
I hope this doesn't break stuff. The definitions for pitch previously were technically incorrect, and counter to industry standards, so if fixing breaks something, then maybe we can work on a a plan of getting it right while maintaining legacy. Let me know your plan as far as accepting or rejecting so I can start work on other improvements.

Reply:
Also, I don't know if I did more harm than good by trying to fix up the tab/space indents to meet the convention.

Reply:
Yeah, in the past it was completely broken, I agree, but at the same time, I see someone's sneaked in a fix to `gear()` already and nobody's complained :laughing:, so at this point I don't really care if we fix it for `rack()` as well and break existing designs.

Reply:
Regarding the tabs/spaces, I'd have preferred that the indentation changes be in a separate commit to keep things cleaner and easier to review, but since you've already done it, I think it's fine